File: 53-MaximumSubarray/53-MaximumSubarray.py
# Last updated: 2/22/2026, 1:55:13 PM
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        s = 0
        mx = nums[0]
        for i in range(len(nums)):
            s+=nums[i]
            if(s>mx): mx = s
            if(s<0): s = 0
        return mx

File: 53-MaximumSubarray/53-MaximumSubarray.java
// Last updated: 2/22/2026, 1:55:05 PM
class Solution {
    public int maxSubArray(int[] nums) {
        int sum = 0, max = Integer.MIN_VALUE;
        for(int num: nums)
        {
            sum+=num;
            if(sum>max) max = sum;
            if(sum<0) sum = 0;
        }
        return max;
    }
}